src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,231 @@
+    return S_OK;
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetInstrumentationMethod(_In_ REFGUID cslid, _Out_ IUnknown** ppUnknown)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IUnknown> pUnknown;
+    IfFailRet(m_pProfilerManager->GetInstrumentationMethod(cslid, &pUnknown));
+    *ppUnknown = pUnknown.Detach();
+    return S_OK;
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IInstrumentationMethod> pInstrumentationMethod;
+    IfFailRet(m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        &pInstrumentationMethod));
+    *ppInstrumentationMethod = pInstrumentationMethod.Detach();
+    return S_OK;
+}
+
+// IProfilerManager2 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    HRESULT hr = S_OK;
+    DWORD apiVer;
+    IfFailRet(m_pProfilerManager->GetApiVersion(&apiVer));
+    *pApiVer = apiVer;
+    return S_OK;
+}
+
+// IProfilerManager4 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pLogging;
+    IfFailRet(m_pProfilerManager->GetGlobalLoggingInstance(&pLogging));
+    *ppLogging = pLogging.Detach();
+    return S_OK;
+}
+
+
+// IProfilerManager5 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    HRESULT hr = S_OK;
+    BOOL fRegistered;
+    IfFailRet(m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, &fRegistered));
+    *pfRegistered = fRegistered;
+    return S_OK;
+}
+
+// IProfilerManagerLogging Methods
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    WCHAR wszPrefix[MAX_PATH];
+    wcscpy_s(wszPrefix, m_wszInstrumentationMethodGuid.c_str());
```

Comment:
I think it would be better if these were more formatted, something like "[IM:00000000-0000-0000-0000-000000000000]..." so that it's easier to know what the GUID is for and easier to parse out this information. I would like to do this in general for most of the information, such as changing the "LogError" and "LogMessage" to a similar format, but that would be beyond the scope of this change.

Reply:
I agree, a better design would be WriteLog(LogLevel, IMGuid, MessageStr) to be more explicit about these formats. It would require a new interface.

---
In reply to: [331297825](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331297825) [](ancestors = 331297825)